src/main/java/uk/gov/justice/digital/writer/Writer.java

```diff
@@ -74,16 +74,18 @@ protected static Consumer<Row> processRow(
             SourceReference.PrimaryKey primaryKey
     ) {
         return row -> {
-            val optionalOperation = getOperation(row.getAs(OPERATION));
+            String unvalidatedOperation = row.getAs(OPERATION);
+            val optionalOperation = getOperation(unvalidatedOperation);
+
             if (optionalOperation.isPresent()) {
                 val operation = optionalOperation.get();
                 try {
                     writeRow(spark, storage, destinationPath, primaryKey, operation, row);
                 } catch (DataStorageException ex) {
-                    logger.warn("Failed to {}: {} to {}", operation.getName(), row.json(), destinationPath);
+                    logger.error("Failed to {}", String.format("%s: %s to %s", operation.getName(), row.json(), destinationPath), ex);
```

Comment:
Hmm...the `{}` interpolation should work here without you having to use a `String.format` like line 88 below.